Content.Server/Chat/TelepathicChatSystem.Psychognomy.cs

```diff
@@ -0,0 +1,189 @@
+using Content.Shared.Humanoid;
+using Content.Shared.Damage;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Physics;
+using Content.Shared.Speech;
+using Content.Shared.Speech.Muting;
+using Content.Shared.CombatMode.Pacification;
+using Content.Shared.CombatMode;
+using Content.Server.Bible.Components;
+using Content.Server.Research.Oracle;
+using Robust.Shared.GameObjects.Components.Localization;
+using Robust.Shared.Enums;
+using Content.Server.Nyanotrasen.Research.SophicScribe;
+using Content.Server.Nyanotrasen.Cloning;
+using Content.Server.Psionics.Glimmer;
+using Robust.Shared.Physics;
+using Robust.Shared.Physics.Components;
+using Content.Shared.Nutrition.Components;
+using Content.Server.Vampiric;
+using Content.Shared.Revenant.Components;
+using Content.Shared.Abilities.Psionics;
+using Content.Server.Abilities.Psionics;
+
+namespace Content.Server.Chat;
+public sealed partial class TelepathicChatSystem
+{
+    private void InitializePsychognomy()
+    {
+        SubscribeLocalEvent<HumanoidAppearanceComponent, GetPsychognomicDescriptorEvent>(DescribeHumanoid);
+        SubscribeLocalEvent<GrammarComponent, GetPsychognomicDescriptorEvent>(DescribeGrammar);
+        SubscribeLocalEvent<DamageableComponent, GetPsychognomicDescriptorEvent>(DescribeDamage);
+        SubscribeLocalEvent<MobStateComponent, GetPsychognomicDescriptorEvent>(DescribeMobState);
+        SubscribeLocalEvent<HungerComponent, GetPsychognomicDescriptorEvent>(DescribeHunger);
+        SubscribeLocalEvent<PhysicsComponent, GetPsychognomicDescriptorEvent>(DescribePhysics);
+        SubscribeLocalEvent<FixturesComponent, GetPsychognomicDescriptorEvent>(DescribeFixtures);
+        SubscribeLocalEvent<MetempsychosisKarmaComponent, GetPsychognomicDescriptorEvent>(DescribeKarma);
+        SubscribeLocalEvent<GlimmerSourceComponent, GetPsychognomicDescriptorEvent>(DescribeGlimmerSource);
+        SubscribeLocalEvent<PsionicComponent, GetPsychognomicDescriptorEvent>(DescribePsion);
+        SubscribeLocalEvent<InnatePsionicPowersComponent, GetPsychognomicDescriptorEvent>(DescribeInnatePsionics);
+        SubscribeLocalEvent<BloodSuckerComponent, GetPsychognomicDescriptorEvent>(DescribeBloodsucker);
+        SubscribeLocalEvent<RevenantComponent, GetPsychognomicDescriptorEvent>(DescribeRevenant);
+        // SubscribeLocalEvent<GlimmerWispComponent, GetPsychognomicDescriptorEvent>(DescribeGlimmerWisp);
+        SubscribeLocalEvent<FamiliarComponent, GetPsychognomicDescriptorEvent>(DescribeFamiliar);
+        // SubscribeLocalEvent<GolemComponent, GetPsychognomicDescriptorEvent>(DescribeGolem);
+        SubscribeLocalEvent<OracleComponent, GetPsychognomicDescriptorEvent>(DescribeOracle);
+        SubscribeLocalEvent<SophicScribeComponent, GetPsychognomicDescriptorEvent>(DescribeSophia);
```

Comment:
Actually, looking at all of these, it just occurs to me that every single one of these(except Bloodsucker) shares a PsionicComponent in common(Even InnatePsionic, because it will EnsureComp psionic). So you could actually for instance store a..

[DataField]
public List<string> PsychognomicDescriptor;
on the PsionicComponent, and then that lets you assign descriptors to all of these entities in yaml, rather than needing to hardcode an event for every unique psionic entity. While it won't solve the need to hardcode in an event interaction for truly unique entities, it'll let you get rid of 8 of these event subscriptions, plus uncountable future event subscriptions.